ndnfs/driver/driver.go

```diff
@@ -0,0 +1,360 @@
+	content, err := ioutil.ReadFile(fname)
+	if err != nil {
+		log.Fatal(DN, "Error reading config file: ", fname, " error: ", err)
+	}
+	var conf Config
+	err = json.Unmarshal(content, &conf)
+	if err != nil {
+		log.Fatal(DN, "Error parsing config file: ", fname, " error: ", err)
+	}
+	return conf
+}
+
+func DriverAlloc(cfgFile string) NdnfsDriver {
+	conf := ReadParseConfig(cfgFile)
+	if conf.Chunksize == 0 {
+		conf.Chunksize = defaultChunkSize
+	}
+	if conf.Mountpoint == "" {
+		conf.Mountpoint = defaultMountPoint
+	}
+	log.Info(DN, " config: ", conf)
+	d := NdnfsDriver{
+		Scope:			"local",
+		DefaultVolSz:	1024,
+		Mutex:          &sync.Mutex{},
+		Endpoint:       fmt.Sprintf("http://%s:%d/", conf.Nedgerest, conf.Nedgeport),
+		Config:         &conf,
+	}
+	return d
+}
+
+func (d *NdnfsDriver) Request(method, endpoint string, data map[string]interface{}) (body []byte, err error) {
+	log.Debug("Issue request to Nexenta, endpoint: ", endpoint, " data: ", data, " method: ", method)
+	if endpoint == "" {
+		err = errors.New("Unable to issue requests without specifying Endpoint")
+		log.Fatal(err.Error())
+	}
+	datajson, err := json.Marshal(data)
+	if (err != nil) {
+		log.Fatal(err)
+	}
+
+	tr := &http.Transport{}
+	client := &http.Client{Transport: tr}
+	url := d.Endpoint + endpoint
+	req, err := http.NewRequest(method, url, nil)
+	if len(data) != 0 {
+		req, err = http.NewRequest(method, url, strings.NewReader(string(datajson)))
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Basic " + basicAuth(d.Config.Username, d.Config.Password))
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Fatal("Error while handling request ", err)
+	}
+	body, err = ioutil.ReadAll(resp.Body)
+	log.Debug("Got response, code: ", resp.StatusCode, ", body: ", string(body))
+	d.checkError(resp)
+	defer resp.Body.Close()
+	if (err != nil) {
+		log.Fatal(err)
+	}
+	return body, err
+}
+
+func basicAuth(username, password string) string {
+	auth := username + ":" + password
+	return base64.StdEncoding.EncodeToString([]byte(auth))
+}
+
+func (d *NdnfsDriver) checkError(resp *http.Response) (err error) {
+	if resp.StatusCode > 399 {
+		body, err := ioutil.ReadAll(resp.Body)
+		log.Error(resp.StatusCode, body, err)
+		return err
+	}
+	return err
+}
+
+func (d NdnfsDriver) Capabilities(r volume.Request) volume.Response {
+	log.Debug(DN, "Received Capabilities req")
+	return volume.Response{Capabilities: volume.Capability{Scope: d.Scope}}
+}
+
+func (d NdnfsDriver) Create(r volume.Request) volume.Response {
+	log.Debugf(fmt.Sprintf("Create volume %s using %s with options: %s", r.Name, DN, r.Options))
+	d.Mutex.Lock()
+	defer d.Mutex.Unlock()
+
+	var cluster, tenant, service string
+	var chunkSizeInt int
+	var err error
+	if r.Options["chunksize"] != "" {
+		chunkSizeInt, _ = strconv.Atoi(r.Options["chunksize"])
+	} else {
+		chunkSizeInt = d.Config.Chunksize
+	}
+
+	if chunkSizeInt < 4096 {
```

Comment:
In nedge there are a few more limitations:
min 4096, max 1048576, power of 2
At this time, if you set any value that does not fit the rules, ganesha may crash. So it is good to validate the chunkSize more thoroughly at client side.